api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -100,35 +104,73 @@ const (
 
 	gacEnvKey  = "GOOGLE_APPLICATION_CREDENTIALS"
 	saFilePath = "/secret/kaniko-secret.json"
+
+	baseImageEnvKey            = "BASE_IMAGE"
+	modelDependenciesUrlEnvKey = "MODEL_DEPENDENCIES_URL"
+	modelArtifactsUrlEnvKey    = "MODEL_ARTIFACTS_URL"
+
+	modelDependenciesPath = "/merlin/model_dependencies"
 )
 
 var (
 	jobTTLSecondAfterComplete int32 = 3600 * 24 // 24 hours
 	jobCompletions            int32 = 1
 )
 
-func newImageBuilder(kubeClient kubernetes.Interface, config Config, nameGenerator nameGenerator) ImageBuilder {
+func newImageBuilder(kubeClient kubernetes.Interface, config Config, nameGenerator nameGenerator, gsutil gsutil.GSUtil) ImageBuilder {
 	return &imageBuilder{
 		kubeClient:    kubeClient,
 		config:        config,
 		nameGenerator: nameGenerator,
+		gsutil:        gsutil,
 	}
 }
 
 // GetMainAppPath Returns the path to run the main.py of batch predictor, as configured via env var
 func (c *imageBuilder) GetMainAppPath(version *models.Version) (string, error) {
-	baseImageTag, ok := c.config.BaseImages[version.PythonVersion]
```

Comment:
I don't think we should remove this check entirely. Can we add a `SupportedPythonVersions []string` field under `ImageBuilderConfig` and use that to check? Ultimately, the pyfunc real time and batch server have some Python version constraints as does the SDK ([example](https://github.com/caraml-dev/merlin/blob/main/python/pyfunc-server/setup.py#L48)). We should capture this in the config and check here to prevent unnecessarily triggering image building for unsupported Python versions.

Reply:
Oops, I forgot about SupportedPythonVersions. I've updated it